src/CapabilityService.WebApi/Application/CapabilityApplicationService.cs

```diff
@@ -13,13 +13,23 @@ public class CapabilityApplicationService : ICapabilityApplicationService
         private readonly ICapabilityRepository _capabilityRepository;
         private readonly ITopicRepository _topicRepository;
 
-        public CapabilityApplicationService(ICapabilityRepository capabilityRepository, ITopicRepository topicRepository)
-       {
+        public CapabilityApplicationService(ICapabilityRepository capabilityRepository,
+            ITopicRepository topicRepository)
+        {
             _capabilityRepository = capabilityRepository;
             _topicRepository = topicRepository;
         }
 
         public Task<Capability> GetCapability(Guid id) => _capabilityRepository.Get(id);
+
+        public async Task DeleteCapability(Guid id)
+        {
+            var capability = await _capabilityRepository.Get(id);
+            capability.Delete();
+            await _capabilityRepository.Remove(capability);
+        }
+
+
         public async Task<Capability> UpdateCapability(Guid id, string newName, string newDescription)
         {
             Capability.Create(newName, newDescription);

```

Comment:
I think we should turn CapabilityName into a value type if we need to validate names. Using the factory method for a Capability to do so will create unused memory allocations which will be GC'ed eventually. On a side note the string in the factory (Create) method should be moved to a .resx file. :)